### ISSUE TITLE
swap-windows check delivers wrong result

### ISSUE THREAD

**Issue by Gammenon** [Icinga/icinga2#5815]
The server on which the problem was detected has its own partition for swap. This partition is 10GB in size.

In the system settings for virtual memory, the size is limited to 10GB initial and maximum.

The pagefile is currently also about 10GB in size.

In Icinga / Icingaweb2 but also with manual execution you get totally different results.

The current value is ~19GB and the maximum possible is said to be ~25GB.

Wrong counters seem to be used here.

![2017-11-28 15_44_11-win-swap_02](https://user-images.githubusercontent.com/34088644/33364281-ff4641e0-d4e3-11e7-98e1-cd6b4de7072b.png)
![2017-11-28 15_37_46-win-swap_01](https://user-images.githubusercontent.com/34088644/33364282-ff882150-d4e3-11e7-97db-863a3048dd81.png)

**Comment by lazyfrosch** [comment]
I had a look in the code, Windows is a bit weird in terms of the API for those values

https://github.com/Icinga/icinga2/blob/f3c825cb86500bfe051f6c6cd0295cf260f70dfb/plugins/check_swap.cpp#L199-L206

Docs on [GlobalMemoryStatusEx](https://msdn.microsoft.com/de-de/library/windows/desktop/aa366589(v=vs.85).aspx) or better [MEMORYSTATUSEX structure](https://msdn.microsoft.com/de-de/library/windows/desktop/aa366770(v=vs.85).aspx) state that:

```
ullTotalPageFile
The current committed memory limit for the system or the current process, whichever is smaller, in bytes. To get the system-wide committed memory limit, call GetPerformanceInfo.

ullAvailPageFile
The maximum amount of memory the current process can commit, in bytes. This value is equal to or smaller than the system-wide available commit value. To calculate the system-wide available commit value, call GetPerformanceInfo and subtract the value of CommitTotal from the value of CommitLimit.
```

So this sounds more like process related than system, but the returned value is clearly wrong.

CommitLimit from [PERFORMANCE_INFORMATION](https://msdn.microsoft.com/de-de/library/windows/desktop/ms683210(v=vs.85).aspx) sounds more useful, but there doesn't seem to be a value about how much is paged right now...

Could @LordHepipud and @Crunsher have a look?

**Comment by Crunsher** [comment]
I remember running into that problem when I first wrote the Plugin (ie being very confused by the documentation).

A workaround is to use check_perfom with the corresponding counter.

**Comment by lazyfrosch** [comment]
Problem seems to be that these counters always include the RAM + PageFile.

So we need to do:

```cpp
 DWORDLONG total, avail;
 total = MemBuf.ullTotalPageFile - MemBuf.ullTotalPhys;
 avail = MemBuf.ullAvailPageFile - MemBuf.ullAvailPhys;

 printInfo.tSwap = round(total / pow(1024.0, printInfo.unit)); 
 printInfo.aSwap = round(avail / pow(1024.0, printInfo.unit)); 
 printInfo.percentFree = 100.0 * avail / total; 
```

**Comment by mcktr** [comment]
While testing the code from @lazyfrosch I ran into following problem: If I run the check after a reboot of my windows machine the check returns a value above 100%.

```
SWAP OK - 101.041% free | swap=5052MB;;;0;5000
```

The page file is set to a fix value of 5000 MB. To preclude a wrong behavior on my windows machine I will test this on another (non-virtual) windows machine. I'll provide feedback once I have tested this.

**Comment by mcktr** [comment]
I did further tests and I always got strange outputs above 100% (on several machines with and without fixed pagefile setted). I also tried to use PERFORMANCE_INFORMATION as suggested, but this also includes RAM + pagefile.

But I found another promising candidate to get the actual pagefile size:
[ENUM_PAGE_FILE_INFORMATION](https://msdn.microsoft.com/en-us/library/windows/desktop/ms682646(v=vs.85).aspx).  This returns the actual page file size without RAM. First tests seems to be good.
